eclipse-jdt/eclipse.jdt.core
# Add test for github issue #3340

## What it does
This adds a test case for https://github.com/eclipse-jdt/eclipse.jdt.core/issues/3340 demonstrating the incorrect behaviour.

## How to test
To reproduce the test failure, run `org.eclipse.jdt.core.tests.model` project with `mvn verify -Ptest-on-javase-21 -Dtest=org.eclipse.jdt.core.tests.formatter.FormatterCommentsBugsTest` under JDK 21

## Author checklist
